pkg/server/slicerouter_config.go

```diff
@@ -244,33 +242,54 @@ func vl3ReconcileRoutesInKernel() error {
 		return err
 	}
 
-	routeMap := make(map[string]netlink.Route)
+	routeMap := make(map[string][]netlink.Route)
 	for _, route := range installedRoutes {
 		// Default route will have a Dst of nil so it is
 		// important to have a null check here. Else we will
 		// crash trying to deref a null pointer.
 		if route.Dst == nil {
 			continue
 		}
-		routeMap[route.Dst.String()] = route
+		routeMap[route.Dst.String()] = append(routeMap[route.Dst.String()], route)
 	}
+	logger.GlobalLogger.Infof("installed Route: %v", installedRoutes)
 	logger.GlobalLogger.Infof("Route map: %v", routeMap)
 	logger.GlobalLogger.Infof("Slice Route map: %v", remoteSubnetRouteMap)
 
-	for remoteSubnet, nextHop := range remoteSubnetRouteMap {
-		_, ok := routeMap[remoteSubnet]
-		// If the route is absent or the nexthop is incorrect, reinstall the route.
-		if !ok || routeMap[remoteSubnet].Gw.String() != nextHop {
-			logger.GlobalLogger.Infof("Installed route does not reflect slice state. Reconciling dst: %v, gw: %v", remoteSubnet, nextHop)
-			err := vl3InjectRouteInKernel(remoteSubnet, nextHop)
-			if err != nil {
-				logger.GlobalLogger.Errorf("Failed to install route: dst: %v, gw: %v", remoteSubnet, nextHop)
+	nextHopInfoSlice := []*netlink.NexthopInfo{}
+
+	for remoteSubnet, nextHopList := range remoteSubnetRouteMap {
+		// avoid injecting routes if route map is empty
+		if len(routeMap[remoteSubnet]) == 0 {
+			break
+		}
+		for _, ip := range nextHopList {
+			_, ok := routeMap[remoteSubnet]
+			// If the route is absent or nexthop is incorrect, reinstall the route.
+			if !ok || containsRoute(routeMap[remoteSubnet], ip) {
+				gwObj := &netlink.NexthopInfo{Gw: net.ParseIP(ip)}
+				nextHopInfoSlice = append(nextHopInfoSlice, gwObj)
+
+				logger.GlobalLogger.Infof("Installed route does not reflect slice state. Reconciling dst: %v, gw: %v", remoteSubnet, nextHopInfoSlice)
+				err := vl3InjectRouteInKernel(remoteSubnet, nextHopInfoSlice)
```

Comment:
Does this func update the routeMap? Once the route is installed, the route map needs to be updated.